WordPressKitTests/Utilities/HTTPHeaderValueParserTests.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+import XCTest
+
+@testable import WordPressKit
+
+class HTTPHeaderValueParserTests: XCTestCase {
+
+    func testReturnOriginalCase() {
+        XCTAssertEqual(
+            HTTPURLResponse.value(ofParameter: "charset", inHeaderValue: "application/json; charset=UtF-8"),
+            "UtF-8"
+        )
+    }
+
+    func testCaseInsensitiveParameter() {
+        XCTAssertEqual(
+            HTTPURLResponse.value(ofParameter: "CharSet", inHeaderValue: "application/json; charset=utf-8"),
+            "utf-8"
+        )
+    }
+
+    func testFirstParameter() {
+        XCTAssertEqual(
+            HTTPURLResponse.value(ofParameter: "CharSet", inHeaderValue: "application/json; charset=utf-8;"),
```

Comment:
Given the test is "first parameter" and we also have a "last parameter" test below with the same input, I'd say the input here should have been swapped?

```suggestion
            HTTPURLResponse.value(ofParameter: "CharSet", inHeaderValue: "charset=utf-8; application/json;"),
```

Reply:
Good catch! The charset here is the first and last parameter. I have updated the other test case to make charset as last parameter only. https://github.com/wordpress-mobile/WordPressKit-iOS/pull/666/commits/44a4fb1d746fcdbc2e35c4e88f3d4d9c85943fdc

I don't think `charset=utf-8; application/json;` is a valid header value. The 'application/json' part is the true header value ('Content-Type'). Other texts followed are parameters, which are additional information.

Reply:
> I don't think charset=utf-8; application/json; is a valid header value.

Good point.

I wonder what's a reasonable behavior for the code in that case 🤔

Reply:
"undefined behaviour" I guess 🥲 .